features/networking/multus.feature

```diff
@@ -416,4 +416,75 @@ Feature: Multus-CNI related scenarios
+	    | macvlan mode bridge |
+  
+  # @author anusaxen@redhat.com
+  # @case_id OCP-21946
+  @admin
+  Scenario: The multus admission controller should be able to detect the syntax issue in the net-attach-def
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin and simulating syntax errors
+    Given I have a project
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml" replacing paths:
+      | ["spec"]["config"]| 'asdf' |
+    Then the step should fail
+    And the output should contain:
+      | admission webhook "multus-validating-config.k8s.io" denied the request|
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge@$ |
+    Then the step should fail
+    And the output should contain:
+      |subdomain must consist of lower case alphanumeric characters|
+
+  # @author anusaxen@redhat.com
+  # @case_id OCP-21949
+  @admin
+  Scenario: The multus admission controller should be able to detect the issue in the pod template
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I have a project
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml |
+    Then the step should succeed
+    # Create a pod consuming net-attach-def simulating wrong syntax in name
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/1interface-macvlan-bridge.yaml" replacing paths:
+      | ["metadata"]["generateName"] | macvlan-bridge-pod-$@ |
+    Then the step should fail
+    And the output should contain:
+      | subdomain must consist of lower case alphanumeric characters |
+
+  # @author anusaxen@redhat.com
+  # @case_id OCP-21793
+  @admin
+  Scenario: User cannot consume the net-attach-def created in other project which is namespace isolated	
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I switch to cluster admin pseudo user
+    And I use the "default" project
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/macvlan-bridge.yaml |
+    Then the step should succeed
+    And admin ensures "macvlan-bridge" network_attachment_definition is deleted from the "default" project after scenario
```

Comment:
my suggestion is to use a name like `macvlan-bridge-21793` or similar to avoid interference with other test scenarios using same name. Tests run in parallel. For creating you could also use step 
```
/^I run oc create( as admin)? (?:over|with) #{QUOTED} replacing paths:$/
```
It allows you to update some part of object before it is created. See existing usage of step for an example.

Reply:
Yep i am aware of  that step but but not sure if this would interfere with other as this is only case in default project and rest other scenario will create it in unique projects. Even if another default one lands in future, they would run sequentially not parallel and would cleanup before exit. Correct if i am wrong :)

Reply:
Taken care of it